packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.spec.ts

```diff
@@ -1302,6 +1309,164 @@ describe('RichTextEditor', () => {
+                            )
+                        ).toBeTrue();
+                        expect(
+                            pageObject.getButtonAriaDisabledValue(
+                                value.toolbarButtonIndex
+                            )
+                        ).toBe('true');
+                    }
+                );
+            }
+        });
+    });
+
+    it('should have footer section hidden when footer-hidden enabled', async () => {
+        expect(pageObject.isFooterVisible()).toBeTrue();
+
+        await pageObject.setFooterHiddenAttribute();
+
+        expect(pageObject.isFooterVisible()).toBeFalse();
+        expect(pageObject.getFooterHeight()).toBe('0px');
+    });
+
+    it('should fire "input" event when there is an input to the editor', async () => {
+        const inputEventListener = createEventListener(element, 'input');
+
+        await pageObject.setEditorTextContent('input');
+        await inputEventListener.promise;
+
+        expect(inputEventListener.spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should fire "input" event when setting the content through "setMarkdown"', async () => {
+        const inputEventListener = createEventListener(element, 'input');
+
+        element.setMarkdown('input');
+        await inputEventListener.promise;
+
+        expect(inputEventListener.spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should fire "input" event when the text is updated/removed from the editor', async () => {
+        const inputEventListener = createEventListener(element, 'input');
+
+        await pageObject.setEditorTextContent('update');
+        await inputEventListener.promise;
+
+        expect(inputEventListener.spy).toHaveBeenCalledTimes(1);
+
+        await pageObject.setEditorTextContent('');
+        await inputEventListener.promise;
+
+        expect(inputEventListener.spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should initialize "empty" to true and false when there is a content', async () => {
+        expect(element.empty).toBeTrue();
+
+        await pageObject.setEditorTextContent('not empty');
+        expect(element.empty).toBeFalse();
+
+        await pageObject.setEditorTextContent('');
+        expect(element.empty).toBeTrue();
+    });
+
+    it('should "empty" return true when the content is loaded with "setMarkdown"', () => {
+        expect(element.empty).toBeTrue();
+
+        element.setMarkdown('not empty');
+        expect(element.empty).toBeFalse();
+
+        element.setMarkdown('');
+        expect(element.empty).toBeTrue();
+    });
+
+    it('should "empty" return true if there is only a whitespace', async () => {
+        expect(element.empty).toBeTrue();
+
+        await pageObject.setEditorTextContent('       ');
+        expect(element.empty).toBeTrue();
+
+        element.setMarkdown('  ');
+        expect(element.empty).toBeTrue();
+    });
+
+    it('should reflect the "placeholder" value to its internal attribute', () => {
```

Comment:
Some more test cases needed? 
1. `empty` should be `true` even when `placeholder` content is set
2. `placeholder` should be cleared when `placeholder` attribute is set to `''` after setting `placeholder`

Reply:
Written both tests as mentioned.